### DISCUSSION TITLE
Allow Someone to Set a Perma-redirect When Changing Username

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#18547] [Feature Requests]
Occasionally, community members change their usernames (or they have an org and change the org's name) and redirects are auto setup on their behalf.

So, for instance I might change my username from https://dev.to/michaeltharrington to https://dev.to/michaeltharrington2 and anyone trying to navigate to https://dev.to/michaeltharrington would instead be taken to https://dev.to/michaeltharrington2. Similarly, if I had links up to my old DEV posts, like https://dev.to/michaeltharrington/i-just-discovered-midjourney-585a if someone tried to navigate to that page post username change, they would land on https://dev.to/michaeltharrington2/i-just-discovered-midjourney-585a.

The issue with this is that someone could come along and take https://dev.to/michaeltharrington/ and that would ruin my redirects. It would be nice if there was a way for folks to set up a perma-redirect so that this couldn't happen. I think we'd want to consider limiting redirects to just one name and potentially limit it to just trusted users to prevent abuse.

This suggestion comes from 2 recent conversations:
- 1 with a user who wants to change their name to start working on their personal brand
- another with an organization that similarly is going through rebranding

**Comment by benhalpern** [comment]
That's a reasonable request.
